### PR TITLE
Fix further cyclical dependencies in actions

### PR DESCRIPTION
**Pull request by hetunandu**
- Move post body switching logic into a saga for better consolidation
- Improve set action property saga to handle object updates

### CODE REVIEW

File: app/client/src/constants/ApiEditorConstants.ts [hunk 1]
```diff
@@ -22,32 +22,41 @@ export const HTTP_METHOD_OPTIONS = HTTP_METHODS.map((method) => ({
...
-  headers: [
-    { key: "", value: "" },
-    { key: "", value: "" },
-  ],
-  queryParameters: [
-    { key: "", value: "" },
-    { key: "", value: "" },
-  ],
+  headers: EMPTY_KEY_VALUE_PAIRS,
+  queryParameters: EMPTY_KEY_VALUE_PAIRS,
```
Comment (rishabhsaxena) [inline app/client/src/constants/ApiEditorConstants.ts:35]: Maybe we should create a new copy here (`EMPTY_KEY_VALUE_PAIRS.slice()`) just as a precaution

File: app/client/src/sagas/ActionSagas.ts [hunk 2]
```diff
@@ -588,6 +599,7 @@ function* setActionPropertySaga(action: ReduxAction<SetActionPropertyPayload>) {
     value,
     propertyName,
   );
+  console.log({ effects });
```
Comment (rishabhsaxena) [inline app/client/src/sagas/ActionSagas.ts:602]: Not sure if this is still needed

File: app/client/src/sagas/ApiPaneSagas.ts [hunk 3]
```diff
@@ -129,6 +131,80 @@ function* syncApiParamsSaga(
...
 }
 
+function* handleUpdateBodyContentType(
+  action: ReduxAction<{ title: ApiContentTypes; apiId: string }>,
+) {
+  const { title, apiId } = action.payload;
+  const { values } = yield select(getFormData, API_EDITOR_FORM_NAME);
+  const displayFormatObject = POST_BODY_FORMAT_OPTIONS.filter(
+    (el) => el.label === title,
+  )[0];
```
Comment (rishabhsaxena) [inline app/client/src/sagas/ApiPaneSagas.ts:141]: Nit: could use `array.find` here?

File: app/client/src/sagas/ApiPaneSagas.ts [hunk 4]
```diff
@@ -129,6 +131,80 @@ function* syncApiParamsSaga(
...
+          displayFormat: POST_BODY_FORMAT_OPTIONS[3],
+        },
+      },
+    });
+    return;
+  }
+
+  const headers = cloneDeep(values.actionConfiguration.headers);
+  const bodyFormData = cloneDeep(values.actionConfiguration.bodyFormData);
+  console.log({ bodyFormData, headers });
```
Comment (rishabhsaxena) [inline app/client/src/sagas/ApiPaneSagas.ts:161]: nit: not sure if needed

**Review by rishabhsaxena — DISMISSED** [review]
Just a few nitpicks

**Review by rishabhsaxena — APPROVED** [review]

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**